packages/pagination/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useIdAllocator } from '@leafygreen-ui/hooks';
+import ChevronLeft from '@leafygreen-ui/icon/dist/ChevronLeft';
+import ChevronRight from '@leafygreen-ui/icon/dist/ChevronRight';
+import IconButton from '@leafygreen-ui/icon-button';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import LeafyGreenProvider from '@leafygreen-ui/leafygreen-provider';
+import { Option, Select } from '@leafygreen-ui/select';
+import { Body } from '@leafygreen-ui/typography';
+
+import { baseStyles, flexSectionStyles } from './Pagination.styles';
+import { PaginationProps } from './Pagination.types';
+
+const DEFAULT_ITEMS_PER_PAGE_OPTIONS = [10, 25, 50];
+
+function Pagination({
+  id: idProp,
+  className,
+  itemsPerPage = DEFAULT_ITEMS_PER_PAGE_OPTIONS[0],
+  itemsPerPageOptions = DEFAULT_ITEMS_PER_PAGE_OPTIONS,
+  onItemsPerPageOptionChange,
+  currentPage = 1,
+  onCurrentPageOptionChange,
+  numTotalItems,
+  onBackArrowClick,
+  onForwardArrowClick,
+  darkMode: darkModeProp,
+  ...rest
+}: PaginationProps) {
+  const { darkMode } = useDarkMode(darkModeProp);
+  const itemsPerPageLabelId = useIdAllocator({
+    prefix: 'lg-pagination-items-per-page-label',
+    id: idProp,
+  });
+  const itemsPerPageSelectId = useIdAllocator({
+    prefix: 'lg-pagination-items-per-page-select',
+    id: idProp,
+  });
+
+  const getCurrentRangeString = () => {
```

Comment:
could these functions accept args and live outside the component itself?

Reply:
They could - I can make that change